app/models/user.rb

```diff
@@ -0,0 +1,23 @@
+class User < ApplicationRecord
+  # constants
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  # validations
+  validates :email, format: {with: VALID_EMAIL_REGEX, message: I18n.t("errors.email_fomat")}
+  validates :email, length: {maximum: 255, message: I18n.t("errors.too_long")}
+  validates :email, presence: {message: I18n.t("errors.cant_blank")}
+  validates :email, uniqueness: {case_sensitive: false, message: I18n.t("errors.taken")}
+  validates :name, length: {maximum: 50, message: I18n.t("errors.too_long")}
+  validates :name, presence: {message: I18n.t("errors.cant_blank")}
+  validates :password, length: {minimum: 6, message: I18n.t("errors.too_short")}
+  validates :password, presence: {message: I18n.t("errors.cant_blank")}
+
+  # callback macro
+  before_save :dowcase_email
+
+  has_secure_password
```

Comment:
Line này khai báo dùng làm gì em? cơ chế hoạt động như thế nào?

Reply:
chỗ này là method của Rails để mã hóa password, confirm với xác thực password đó chị.
để sau này thực hiện task sign-up với log-in
@dothidiemthao

Reply:
@dothidiemthao 
Chỗ validate em dùng symbol vậy được không chị?